src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -358,6 +367,9 @@ ContinuousBatchingPipeline::ContinuousBatchingImpl::generate(const std::vector<o
         }
 
         result.m_status = generations[request_id]->get_status();
+        
+        // The same perf metrics for each sequence, only tokenization/detokenization will differ.
+        result.perf_metrics = perf_metrics;
```

Comment:
`generate_durations` is missed here

`std::vector<EncodedGenerationResult>
ContinuousBatchingPipeline::ContinuousBatchingImpl::generate` should return valid perf metrics.